### PR TITLE
Adds a new Space Ruin, the Hell Factory.

### PR DESCRIPTION
**Pull request by ArcaneMusic**
Working title because the only thing more hellish than poor working conditions is mapping /s.

## About The Pull Request

FIRST AND FOREMOST: Re-adds #49018 because I managed to pull a doublefuckup and reverted it silently in #49084 by accident. I still have no idea how I did that without triggering a single line change. Anyways.

Adds a new Space ruin to the game, the Hell Factory. The Hell Factory uses the new mapping tools from the aforementioned first PR in order to create an interesting puzzle pattern for how crew members need to access and explore this ruin to get at the center prize room, the boss man's office. The map features gratuitous use of holoboxes and hologrids, with a pattern showing off  to players roughly how they work, and teaching players about the new keycards and keycard doors. Additionally, most of the "challenge" here is just efficient space and storage usage in order to get to the center room.  2 of the holoboxes to fully complete the ruin requires the player to enter the freezer in the top left corner of the map, which can very quickly deal exceptionally high damage.

The reward for the ruin is a in the hidden cache in the boss's office, as well as some unique new plumbing synthesizers that allow for the production of blood, honey, and universal enzyme. By itself it allows for the production of Red Mead, but if you want to go to the effort of hauling one of these three all the way back to the station you can open up some interesting plumbing setups. I'm more than willing to consider a different recipe for the alcohol brewing as well if it proves problematic.
(Note: You can already duplicate blood with roundstart plumbing equipment, so my main concern is in the honey production.)

## Why It's Good For The Game

Well, I was already considering how I was going to use the puzzle tools from my earlier PR, then realized that I actually fucked up the entire process, so we're hitting 2 birds with one oddly boomerang shaped stone.

The Map itself lends itself to a theme me a few other people started to work on before this, which is finding a fully automated factory running in the middle of space. The scope and/or nature of the ruin could be expanded upon easily, it's just a matter of maybe cleaning up some of the space use/efficiency further.

- [x] Verify that the ruin's atmos/area tiles don't have any issues.
- [x] Throw in some basic fluff papers around the ruin for fun/glory.
- [x] Come up with a slightly less lame name?

## Changelog
:cl:
add: A new space ruin has been located in your sector. The previous owners have dubbed the place "The Hell Factory", despite the region being designated as an alcohol bottling facility.
fix: Fixes accidentally reverting the puzzles/trapmaking tools PR.
/:cl:

### CODE REVIEW

**Comment by GravitronBlue** [comment]
Might be a missing keycard reward with the third hologrid next to the office door. Currently only rewards you with another cookie.

Key

**Comment by FantasticFwoosh** [comment]
Wouldn't it be a 'Heck Distillery'? Nice to see pressure plates being used.

File: code/datums/ruins/space.dm [hunk 1]
```diff
@@ -301,3 +301,10 @@
 	description = "Well wish you luck."
 	allow_duplicates = FALSE
 	unpickable = TRUE
+
+/datum/map_template/ruin/space/hellfactory
+	id = "hellfactory"
+	suffix = "hellfactory.dmm"
+	name = "Hell Factory"
+	description = "An abandoned warehouse and brewing facility, which has been recently abandoned. Reports claim that the security system entered an ultra-hard lockdown, but these reports are inconclusive."
+
```
Comment (itseasytosee) [inline code/datums/ruins/space.dm]: An abandoned warehouse... which has been recently abandoned

**Comment by itseasytosee** [comment]
The big loot here is a rapid service fab?

**Comment by actioninja** [comment]
Is this ready?

**Pull request merged**